### DISCUSSION TITLE
Any way to remove the resource automatically when argo finished with main container failed?

### DISCUSSION THREAD

**Discussion by jazka** [argoproj/argo-workflows#8366] [General]
# Summary

We created workflow with one step of resource Pytorchjob. When argo main container fails with error like "executor error: Unable to connect to the server: dial tcp xx.xx.xx.xx:443: i/o timeout", the Pytorchjob keeps running, but the workflow will be changed to Error status. Is there any way to remove Pytorchjob and its running pods? The "successCondition" and "failureCondition" have been configed, but it is normal from Pytorchjob. 

# Use Cases

We want to check workflow status to do something, but currently the status of workflow is not accurate to real job.

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

**Comment by ajkaanbal** [comment]
have you tried?

```yaml
setOwnerReference: true
```
and 

```yaml
 podGC:
    strategy: OnWorkflowCompletion
```

https://github.com/argoproj/argo-workflows/blob/master/examples/k8s-owner-reference.yaml
https://github.com/argoproj/argo-workflows/blob/master/examples/pod-gc-strategy.yaml

**Comment by jazka** [comment]
I will try podGC since `setOwnerReference: true` has been set.
